config/environments/test.rb

```diff
@@ -30,6 +30,7 @@
   # The :test delivery method accumulates sent emails in the
   # ActionMailer::Base.deliveries array.
   config.action_mailer.delivery_method = :test
+  config.action_mailer.default_url_options = { host: "localhost", port: 3000 }
```

Comment:
breaking out the hash in this config onto separate lines might be substantially more readable.